### ISSUE TITLE
file load

### ISSUE THREAD

**Issue by HyunjiEllenPak** [AiHackathon2018/AI-Vision#244]
**요청사항**
현재세션에서 데이터를 전처리 후에  .npy로 저장한 후 다음 세션에서도 이를 사용하고 싶습니다. 
#77 의 질문답변 내용을 보니 데이터를 전처리후에 저장한 후  다른 세션에서 로드하여 사용하는 것이 가능하다고 답변주신 것 같은데 데이터를 전처리하고 저장한 후에 다른 세션에서 접근(=로드)이 되지 않습니다.
어떻게 접근해서 로드해야하나요?

**Comment by nsml-admin** [comment]
안녕하세요.

#77 처럼 cache 함수를 쓰셨는데 해결을 안되신 것으로 파악됩니다.
접근을 하지 못하는 상황의 에러로그와 세션 이름, 대략적인 파일의 사이즈를 알려주시면 확인해보겠습니다.

**Comment by HyunjiEllenPak** [comment]
@nsml-admin 현재 세션에서 
nsml.cache(train_data_loader,data_path=train_dataset_path, image_size=input_shape[:2], output_path=output_path) 를 써서 train_data_loader안에서 이미지를 읽어와서 .npy파일로 저장을 하고 세션을 종료한 후 다음 세션에서 로드해서 사용하려고 np.load(output_path[0])을 사용하였는데 해당경로(=output_path[0])가 존재하지 않는다며 로드가 되지 않습니다. 

세션 127번에서 img_list.npy, label_list.npy를 저장하였습니다. 이렇게 저장한 파일의 경우에 어떻게 로드해야하는지 가이드주시면 감사하겠습니다!

**Comment by nsml-admin** [comment]
@HyunjiEllenPak 
load할때도 nsml.cache()를 사용해서 로드해야합니다.

nsml.cache()는 말그대로 캐시기능이라서 처음에 save할때만 실행되고 다음부터는 실행되지않고 캐싱된 파일을 다운받는 역할을 합니다.

예를들어서 코드를 다음과 같이 바꿔주시면 될것같습니다.

```python
nsml.cache(train_data_loader_v0, data_path=train_dataset_path, img_size=input_shape[:2],
               output_path=output_path)
images = np.load(output_path[0])
labels=np.load(output_path[1])
```

**Comment by HyunjiEllenPak** [comment]
@nsml-admin 답변 감사합니다. 
저 말씀해주신대로 load해보기 위해서 load전에,  세션 133번에서 train_data_loader안에서 np.save로 .npy파일을 저장하려다 에러가 발생하였습니다. 에러 원인을 알려주실수 있을까요?

**Comment by HyunjiEllenPak** [comment]
@nsml-admin 
nsml.cache(train_data_loader_v0, data_path=train_dataset_path, img_size=input_shape[:2],
               output_path=output_path)
images = np.load(output_path[0])
labels=np.load(output_path[1])

추가로, 말씀해주신대로 .npy를 저장하고 load를 위와 같이하게 되면 load할때 이전에 저장했던 .npy파일을 가져오는 것이 아니라 다시 train_data_loader_v0에서 파일 cv.imread를 해서 다시 .npy를 만드는 과정을 새롭게 합니다.
cache기능이 뭔가 잘못 된 것같은데, 확인부탁드립니다

**Comment by nsml-admin** [comment]
@HyunjiEllenPak 
OSError는 큰사이즈의 데이터를 만들다가 에러가난거같은데요, 데이터사이즈를 작게 하면 해결될것같습니다.

캐시관련된 문제는 세션을 동시에 여러개를 실행해서 생긴 문제로 보이는데요, 세션 한개에서 cache를 실행하고 cache로 입력된 함수의 실행이 정상적으로 종료후에 다른세션에서 다시 하면 캐싱이 될것같은데요,

혹시 지금 테스트를 해보실수있나요?
